tests/__init__.py

```diff
@@ -0,0 +1 @@
+"""Test the generation of the cookiecutter template."""
```

Comment:
Why needed?

Reply:
Right but our tests is not a public package.

Reply:
The race that they aren't `_tests` means that they are?

Reply:
@matt-graham you know this stuff better than me

Reply:
I'd agree with Sam that we ideally don't want a `__init__.py` in `tests` as we my feeling is that we don't want it to be considered a package, as we'd not usually want to import test modules directly, but as `tests` will still be importable as a namespace package irrespective does make this all a bit ambiguous. [Perhaps more usefully, Pytest's suggested directory structure when having tests separate from main package doesn't include an `__init__.py`](https://docs.pytest.org/en/stable/explanation/goodpractices.html#tests-outside-application-code) so probably good to follow that? We possibly also should then follow the recommendation to add `"--import-mode=importlib"` to `tool.pytest.ini_options.addopts` in `pyproject.toml` to [avoid issues with test modules with non-unique names](https://docs.pytest.org/en/stable/explanation/goodpractices.html#which-import-mode).

Reply:
I missed that this is now closed and conversation is going on elsewhere so ignore above.

Reply:
My bad. Poor thread discipline, and poor redirection.
→ https://github.com/UCL-ARC/python-tooling/pull/219#issuecomment-1794687931 ff

Reply:
I'm going to add an issue for @matt-graham's suggestion!